### PR TITLE
Make search popup dismissible

### PR DESCRIPTION
**Pull request by SiAdcock**
## What does this change?

Currently, it is not easy to dismiss the search popup once it has been opened. The only way to dismiss it is to click the "Search" link a second time.

With this change, the user will be able to dismiss the search popup by clicking anywhere on the page (other than the search popup itself)

I have also removed the `checkResults()` call, as this method is [no longer with us](https://github.com/guardian/frontend/pull/11975). This issue has generated about 1000 errors in Sentry

## What is the value of this and can you measure success?

This makes the search feature more accessible as it is easier to dismiss.

## Does this affect other platforms - Amp, Apps, etc?

No

picture 185

## Tested in CODE?

Oh gosh, no

### CODE REVIEW

File: static/src/javascripts-legacy/projects/common/modules/navigation/search.js [hunk 1]
```diff
@@ -34,10 +34,33 @@ define([
             });
 
             bean.on(document, 'click', '.js-search-toggle', function (e) {
-                searchLoader();
+                var searchToggleLink = $('.js-search-toggle');
+                var searchPopup = $('.popup--search');
```
Comment (gtrufitt) [inline static/src/javascripts-legacy/projects/common/modules/navigation/search.js]: Dunno if you can add a `js-` to this? We should totals lint that 🤔

**Review by NataliaLKB — APPROVED** [review]
Great 🎉 ! Other than Gareth's comment, this looks good to me.

**Comment by paperboyo** [comment]
Wow!!! 💎! Finally (only 2 years): https://github.com/guardian/frontend/issues/7845 (pt. 1), https://github.com/guardian/frontend/issues/3437#issuecomment-37566527, https://github.com/guardian/frontend/issues/7991#issuecomment-71533847. Hugest thanks, @SiAdcock!

**Comment by ScottPainterGNM** [comment]
Yessssssss!

**Pull request merged**